test/strings/types.jl

```diff
@@ -132,10 +161,12 @@ end #let
 
 #for isvalid(SubString{String})
 let s = "Σx + βz - 2"
-  for i in -1:length(s)+2
-      ss=SubString(s,1,i)
-      @test isvalid(ss,i)==isvalid(s,i)
-  end
+    for i in -1:length(s)+2
+        if isvalid(s, i)
+            ss=SubString(s,1,i)
+            @test isvalid(ss,i)==isvalid(s,i)
```

Comment:
`==isvalid(s,i)` is redundant since that's the condition. Also check that a `BoundsError` is raised when `!isvalid(s,i)`.